### ISSUE TITLE
Notify users about changes to map services

### ISSUE THREAD

**Issue by mitchellryant** [TNRIS/dataHub#287]
Next week the migration of the ArcGIS Services and WMS services should be complete and DataHub fully updated. I thinking we will need to sort of "banner" on the Collection Card (where wms link is not null). Simply stating services have been migrated to the a new URL and old URL will be active till Jan 1, 2021. They should update there links.

Since this links do not appear on tnris.org, i do not believe anything need to go there.

__JH edit__: plans to address this issue below --> https://github.com/TNRIS/dataHub/issues/287#issuecomment-724742484

**Comment by jwhaney** [comment]
I think I understand what you're saying here @mitchellryant, but let me play devil's advocate. If the user has the service url link they want, what is their reason to go back to that collection on the datahub? In other words, how will a banner on that collection telling them that the service url is changing help them if they don't have a reason to visit that datahub collection? And when they do visit the datahub collection again after their service url breaks, then they are already there and they can grab the updated url. Just my thoughts anyway, and my thoughts do not always make sense...

I'm thinking this might be a better email blast or tweet maybe? Idk. I don't have the answer necessarily.

**Comment by adambreznicky** [comment]
banner is a good idea for those that do visit. i like the tweet and email blast ideas also. we should do all of them!
maybe could be a blanket banner opposed to only being present on collection cards with a wms link? could work on tnris.org also

should we also attempt to update the old domain routing in Rte53 to just CNAME, or forward traffic to the new domain so that the old ones don't break?

**Comment by jwhaney** [comment]
> should we also attempt to update the old domain routing in Rte53 to just CNAME, or forward traffic to the new domain so that the old ones don't break?

i like this idea. when in doubt, automate the fix!

**Comment by mitchellryant** [comment]
@jwhaney You could do an email blast or twitter, but that would mean i would have to talk to them. and talk about what was done. not really my thing. they would probably make a big deal out of nothing. I can publish the last service and turn the EC2 off. I was just hoping to afford question from staff and customers. Given the lack of problem solving skills SM will not check to see if the URLs actually. 

@adambreznicky i do not think changing route 53 will work. the old site uses a load balance and the services links are slightly different.

**Comment by mitchellryant** [comment]
The use of the email blast or Twitter implied the people using the links are in both of those spheres.

**Comment by jwhaney** [comment]
talked to FR yesterday about this. the plan is to do all three:

1. modal popup notification across entire data.tnris.org domain so any url visited, the notification will show explaining the migration of services, timeline, and how to remedy (general instructions, like visit the dataset/collection and copy the new service url, etc)
2. email blast using mailchimp (@Ldsmiths might be involved with helping on this)
3. tweet link couple sentences with a link to the mailchimp email page

**Comment by mitchellryant** [comment]
i would use the full path to the services directory `https://feature.tnris.org/arcgis/rest/` and `https://imagery.tnris.org/server/rest/`. the other link will return an IIS error.

This is what we said last time. August 25, 2017

```
Notice: Effective August 31, all online map services will migrate to a different cloud provider. Old URLs will not work anymore. The most current URLs are on this page. Read more here: Major Changes and Improvements to TNRIS Online Map Services

Major Changes and Improvements to TNRIS Online Map Services
by IS TEAM | August 25, 2017
Effective August 31, all online map services will migrate to a different cloud provider.

What this change means to you:

- The URL for all raster and vector services will change.
- Web services URLs will be consolidated.
- Web services will continue to be hosted as both ArcGIS and WMS services.
- Fewer Orthoimagery web mapping services will be available, however they will continue to be available through our Data Search & Download.
- All NAIP, TOP, and StratMap image datasets will continue to be offered as web map services.

The benefits to you:

> Increase in speed and image quality.
> Increase in number of vector services published.

Also, we’re investigating the potential to publish elevation data services. Stay tuned for this update!
```

**Comment by mitchellryant** [comment]
the internet wayback machine is really slow.

**Comment by jwhaney** [comment]
> i think it looks good, i would use the full path to the services directory `https://feature.tnris.org/arcgis/rest/` and `https://imagery.tnris.org/server/rest/`. the other link will return an IIS error.

ya, github automatically makes those domains a link but the intent was not to make them links in the notification. i fixed it in my comment. if we include the services directory you mentioned in the comment above, that's only for arcgis connections? what about wms?

**Comment by adambreznicky** [comment]
only thing i'd add is a note of suggestion:
"TNRIS strongly encourages users to switch to update their service connections to the new domain as soon as possible to avoid an unexpected service outage after January 5th"
[10:34 AM]
something along those lines, obviously doesn't need to be those words exactly. i ain't the best linguist

**Comment by adambreznicky** [comment]
since the survey is getting turned off in data.tnris.org, that popup modal could be used for this notification opposed to a new, separate banner component. @ctrepka designed that modal in an agnostic way to serve this type of situation precisely

**Comment by jwhaney** [comment]
> only thing i'd add is a note of suggestion:
> "TNRIS strongly encourages users to switch to update their service connections to the new domain as soon as possible to avoid an unexpected service outage after January 5th"
> something along those lines, obviously doesn't need to be those words exactly. i ain't the best linguist

updated - https://github.com/TNRIS/dataHub/issues/287#issuecomment-724742484

**Comment by mitchellryant** [comment]
everything looks good to me.

**Comment by mitchellryant** [comment]
we can not solve this here but once the old server goes away we will have to remind Staff that this happened. The questions will go directly to Imagery POC, I believe that they all go to RDC.

**Comment by jwhaney** [comment]
related to the datahub survey and adam's [comment](https://github.com/TNRIS/dataHub/issues/287#issuecomment-724843342) about taking it down in time to address the user notification issue here with the same survey template in the api, is that actually doable? seems to me we would need to change some things or build a separate more simplified version for text notifications. for example, the current template has required fields for the form in the Preview Modal section that do not apply to a more simplified notification to users. also, do we manage the simple notification text in SurveyJS? because there is no option to enter simple text in the api form. i may be missing something, but seems for this particular purpose we may need a new simple modal notification setup, but i'm totally open to ideas.

@ctrepka, @JasonKleinert, @adambreznicky please let me know yalls thoughts on this when you have time. thanks.

**Comment by ctrepka** [comment]
@jwhaney Yeah, the way it's currently set up, the apiModal might not work for a simple, single window alert. However, with some minor adjustments to the front-end logic, we could probably make it work well for our needs. For example, we might set it so that if there is not a survey in the full modal, and there is not text or a header for the full modal, the preview modal is the only popup to appear and goes directly to closed state instead of minimized when exited. We could talk about this more in detail (particularly about where the alert content would be managed, in the api or surveyJS), but I think a minor reworking of the current modal solution would make this possible so we don't have to make an entirely separate component.

**Comment by JasonKleinert** [comment]
Just throwing this out there. We used to have a beta notice when we first launched the site. That code is still in the catalog component. We could easily reuse this if a modal is what y'all want. This modal comes up everytime the page is loaded, it does not have any connection to local storage. It may be more appropriate to have a banner at the top of the page for the note about the service urls, but that is up to you. Here's what that modal looks like:
![image](https://user-images.githubusercontent.com/5649664/99113159-62fcae80-25b4-11eb-88ff-f69ab569eb35.png)
